### PR TITLE
Modify periodic test script for v2 branch

### LINKED ISSUE #5411: ODO V2 testing during V3 development
/kind tests
## Which functionality do you think we should add?
Identify tests that are required to be tested during when supporting V2 during V3 development.  

- [x] Identify tests that can be run periodically (low risk) - V2 Branch
- [x] Identify tests that need to be run on a regular basis - V2 Branch

## Why is this needed?
Avoid bottlenecks do to PRs submitted against the two branches and avoid unnecessary costs.

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this:**

/kind tests

**What does this PR do / why we need it:**
This PR will update the script used for periodic testing on openshift-ci.

**Which issue(s) this PR fixes:**

Fixes #5411 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by anandrkskd** [comment]
all so require PR https://github.com/openshift/release/pull/26125 to go in.

File: scripts/openshiftci-periodic-tests.sh [hunk 1]
```diff
@@ -5,6 +5,41 @@ set -e
 # show commands
 set -x
 
+# check if devfiles are updated and if so, update and test them
+check_devfile() {
+    NOTEQUAL= false
```
Comment (feloy) [inline scripts/openshiftci-periodic-tests.sh:10]: This is not an affectation, but this is running the `false` command by defining the env variable NOTEQUAL being equals to the empty string (like you would run `DEBUG=true mycommand`)

(shell is very sensitive to spaces :( )

**Review by feloy — CHANGES_REQUESTED** [review]

File: scripts/openshiftci-periodic-tests.sh (hunk 1 above)
Comment (anandrkskd) [reply scripts/openshiftci-periodic-tests.sh]: Updated!

File: scripts/openshiftci-periodic-tests.sh [hunk 2]
```diff
@@ -5,13 +5,46 @@ set -e
...
+        TEMPDIR=$(mktemp -d)
+        # download devfiles with odo
+        if [[ $LANGUAGE == "springboot" ]]; then
+            odo create java-$LANGUAGE language --context $TEMPDIR
+        else
+            odo create $LANGUAGE language --context $TEMPDIR
+        fi
+
+        Devfile_path=$TEMPDIR/devfile.yaml
+        set +e
```
Comment (feloy) [inline scripts/openshiftci-periodic-tests.sh:25]: Do we need this?
`false || NOTEQUAL="true"` is not a failing command

You would need it to do smth similar to:
```
set +e
diff $Devfile_path $Example_devfile_path 
set -e
if [[ $? == 1 ]]; then
 ...
fi
```

but with the `||`, you are "catching" the error and this is the result of `NOTEQUAL="true" which applies, which is always OK

File: scripts/openshiftci-periodic-tests.sh [hunk 3]
```diff
@@ -5,13 +5,46 @@ set -e
...
+        # check if devfiles differ then the one in examples dir
+        diff $Devfile_path $Example_devfile_path || NOTEQUAL="true" 
+        set -e 
+        if NOTEQUAL="true"; then
```
Comment (feloy) [inline scripts/openshiftci-periodic-tests.sh:29]:
```
if [ "$NOTEQUAL" == "true" ]; then
```

**Review by feloy — CHANGES_REQUESTED** [review]

File: scripts/openshiftci-periodic-tests.sh (hunk 2 above)
Comment (anandrkskd) [reply scripts/openshiftci-periodic-tests.sh:25]: Updated!

**Pull request merged**